frontend/test_runner.go

```diff
@@ -41,6 +41,14 @@ func RunTests(ctx context.Context, client gwclient.Client, spec *dalec.Spec, ref
 		return nil
 	}
 
+	if err := ref.Evaluate(ctx); err != nil {
```

Comment:
Since we're already doing `ref.Evaluate` after we create package build state for `deb`/`rpm`, really the only extra errors this will surface here would be install errors from installing the package in the target container, correct? That should in general be the only extra state that hasn't yet been solved at this point?

Reply:
At this point, the package build state is only evaluated on the packaging build targets, not the container targets.
But other than that, yes you are correct.

Reply:
Ah got it, the build state hasn't yet been evaluated if we've reached this code path from a container target. And in the case of packaging build targets, this `ref.Evaluate` would actually surface errors from installing in the **test** container, not the **target** container since there is no target container in that case. Thanks for clarifying